### DISCUSSION TITLE
no_std support

### DISCUSSION THREAD

**Discussion by dzmitry-huba** [tikv/raft-rs#517] [Ideas]
Hello, Everyone

I was wondering if the maintainers of this project would be open to consider a no_std support for the raft-rs. I did an exercise in a fork to make changes to the codebase to make it no_std compilable (not production ready [draft PR](https://github.com/tikv/raft-rs/pull/516)) and I would like to get feedback on the idea before preparing the official PR. 

Please let me know if no_std support would be welcome. The main scenario is running Raft instance inside of the Trusted Execution Environments. 

Thank you!

**Comment by BusyJay** [comment]
no_std is not a goal but nice to have. It should be OK as long as it doesn't change the std code behavior.

**Reply by dzmitry-huba** [reply]
Thank you, I totally agree with the goal. Let me redo the PR such that no_std is an optional feature while preserving the original std behavior.
